pkgs/servers/nosql/influxdb2/default.nix

```
@@ -0,0 +1,125 @@
+{ buildGoModule
+, buildGoPackage
+, fetchFromGitHub
+, git
+, go-bindata
+, lib
+, llvmPackages
+, mkYarnPackage
+, pkg-config
+, rustPlatform
+, ... }:
+let
+  version = "2.0.2";
+  libflux_version = "0.95.0";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "influxdata";
+    repo = "influxdb";
+    sha256 = "1vv0f5zpxl7s6rsadhsj25zxk04m7a0jzghrbzynjv7xv7ml90r7";
+    # The build process requires `git rev-parse HEAD` to work, so we
+    # ask for submodules (to force the use of fetchGit) and tell it to
+    # leave .git in place. A shallow clone is fine though.
+    fetchSubmodules = true;
+    leaveDotGit = true;
+  };
+
+  ui = mkYarnPackage {
+    src = src;
```

If the ui folder is self-container, you could use:
```suggestion
    src = "${src}/ui";
```
and avoid a bunch of complexity below

---

Unfortunately it's not self contained, it references Swagger definitions in the rest of the codebase. I tried to build with `${src}/ui` originally and it won't work :(